### PR TITLE
Restructure runtime package, and refactor store package 

### PR DESCRIPTION
**Pull request by guowenjian90**
create cri package in runtime package
and move cadvisor into runtime package also.
store refactory
add list object method into store interface,
and use pod/session object definition from its own package in store
add a unit test for store/factory package

### CODE REVIEW

File: pkg/nodeagent/runtime/cadvisor/cadvisor.go [hunk 1]
```diff
@@ -145,40 +154,29 @@ func (cc *cadvisorInfoProvider) Start() error {
...
-							// remember it and remove closed channel after loop
-							panicReceivers[n] = true
-						}
+			event, err := cc.collectCAdvisorInfo()
+			if err == nil {
+				cc.nodeCAdvisorInfo = event
+
+				panicReceivers := make(map[string]bool)
+				for n, r := range cc.receivers {
+					klog.Infof("send node cavisor info to receiver %s", n)
```
Comment (h-w-chen) [inline pkg/nodeagent/runtime/cadvisor/cadvisor.go:164]: nit: cadvisor

File: pkg/nodeagent/store/factory/storefactory_test.go [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+			Status:     v1.PodStatus{},
+		},
+		ConfigMap:        v1.ConfigMap{},
+		RuntimePod:       cri.Pod{},
+		RuntimePodStatus: cri.PodStatus{},
+	}
+	return &testPod
+}
+
+func TestNewPodSqliteStore(t *testing.T) {
```
Comment (h-w-chen) [inline pkg/nodeagent/store/factory/storefactory_test.go:74]: kudos for having necessary unit tests:)

File: pkg/nodeagent/runtime/cri/interface.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+package cri
+
+import (
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/runtime"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	criv1 "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type RuntimeService interface {
```
Comment (h-w-chen) [inline pkg/nodeagent/runtime/cri/interface.go:26]: what is the relation of this interface to the CRI runtime service?

File: pkg/nodeagent/pod/types.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	PodStateRunning       PodState = "Running"
+	PodStateEvacuating    PodState = "Evacuating"
+)
+
+type Container struct {
+	Identifier      string              `json:"identifier,omitempty"`
+	Container       v1.Container        `json:"container,omitempty"`
+	RuntimeConainer cri.ContainerStatus `json:"runtimeConainer,omitempty"`
+}
+type Pod struct {
```
[the remark this answers is not in the document]
Comment (guowenjian90) [reply pkg/nodeagent/pod/types.go:44]: This Pod including cri runtime information, and combine other control plane resources into one, eg. config map and fornax core.
also PodState are specific fornax node agent info.

File: pkg/nodeagent/runtime/cri/interface.go (hunk 3 above)
Comment (guowenjian90) [reply pkg/nodeagent/runtime/cri/interface.go:26]: this is actually cri runtime service interface.
old runtime interface in runtime/interface.go is repurposed to generalize as a interface of runtime dependency of node agent. including cadvisor, cri,  etc. in future to add cpu/memory resource admission and storage.

**Review by h-w-chen — APPROVED** [review]
lgtm

**Pull request merged**